fedbiomed/researcher/federated_workflows/_experiment.py

```diff
@@ -737,12 +737,18 @@ def run_once(self, increase: bool = False, test_after: bool = False) -> int:
         if missing:
             raise FedbiomedExperimentError(ErrorNumbers.FB411.value + ': missing one or several object needed for'
                                            ' starting the `Experiment`. Details:\n' + missing)
+        
         # Sample nodes for training
-
         training_nodes = self._node_selection_strategy.sample_nodes(
             from_nodes=self.filtered_federation_nodes(),
             round_i=self._round_current
         )
+        if not training_nodes:
```

Comment:
On second thoughts, it might be better to be more explicit in the check:
```suggestion
        if len(training_nodes) == 0:
```